Mathlib/Tactic/Linarith/Preprocessing.lean

```diff
@@ -212,7 +212,7 @@ If `pf` is a proof of a strict inequality `(a : ℤ) < b`,
 and similarly if `pf` proves a negated weak inequality.
 -/
 def mkNonstrictIntProof (pf : Expr) : MetaM Expr := do
-  match (← inferType pf).getAppFnArgs with
+  match (← whnfR (← inferType pf)).getAppFnArgs with
```

Comment:
I changed this and the other instance into `instantiateMVars` for the sake of minimalism, and because that seems to more closely go with the style in the rest of the file.

There may well be inconsistencies or bugs in `getAppFnArgs` in the rest of the file, but I think fixes for them should belong in a separate PR.